README.md
# Lofful's WoW Classic Addons
![If you are looking for WoW WOTLK (Click here)](https://github.com/ZaxLofful/Lofful-s-WoW-WOTLK-Addons)

I dual-box as Rogue/Priest (compltely manualy not software asisted in anyway). It can become a hassle to update the addons on both of my computers. So I created this repo initally for that, but some friends wanted my addons as well.

This is an autoupdated list of the addons I use in game. Lots of good stuff...

**List of Addons and Mini-Tutorial for Each:**

**Auctioner** - Basically An Auto-Auction Addon

When you open the auction house it adds extra tabs that are very usefull and will auto-calc prices and stacks for you. Super easy to make money this way. You can scan the whole AH, whic the "play" button. It will save prices for everything so you can review it when out in the field.

**ZPERL** - UI Addon that makes everything moveable and themeable. I use this to arange my addons party portraits (including my own) into a better view.

I personally like to have my chracter portrait right next to my character in the middle of the screen. That way I can see my health and my enemies health as a side by side comparison without looking away.

**FasterLooter** - This mod automatically loots everything from a body and doesn't even show you the loot screen. (I persoanlly don't like seeing the loot screen unless I have to)

You can either disable this mod all together if you like to see the loot window, or you can do Shift+Right-click for it to not auto-loot and show you the loot window.

**WIM** - WoW Instant Messenger - This addon acts like a normal IM client, but its for your whispers. It also will automtically disappear when you are in combat to save screen space.

It makes a pop-up window that allows you to manage your conversations with people via tabs and not having to scroll back through the chat window. It makes whipsers a lot easier to manage and review.

**Usage:**

/zperl in the chat window: This allows you to move everything around and will bring up an extra customziation window to change things even more.

You can turn each of the ZPERL modules on/off independantly through the "All" tab (this is within the extras window)

When using the new Auctioner, just allow it to do a full auto-scan and then cum.

**Release Notes:**

I took out Guide Ally Routes, because it was messing up during QA Unit testing.


![Screen Shot of the latest look open](https://github.com/ZaxLofful/Lofful-s-WoW-Classic-Addons/blob/master/1080p_open.png)


OmniCD/CHANGELOG
------------------------------------------------------------------------------
-- Dragonflight
--
v10.1.7.2768
	Patch 10.1.7 spell updates
	PRIEST Angel's Mercy has been redesigned – Now reduces the cooldown of Desperate Prayer by 20 seconds.
	AUGUST 21, 2023 Hotfixes - Dead of Winter now increases the cooldown of Remorseless Winter by 10 seconds (was 25 seconds).

v10.1.5.2767
	AUGUST 7, 2023 Hotfixes - San’layn now reduces the cooldown of Vampiric Embrace by 30 seconds (was 45 seconds)
	AUGUST 3, 2023 Hotfixes - Fixed an issue where Flow State would not increase the cooldown recovery rate of Emerald Communion.
	Undulating Sporecloak will correctly show for the player and non-synced units
	Casting Fire Blast will correctly reduce the cooldown of cc abilities with Time Manipulation

v10.1.5.2765
	Undulating Sporecloak will correctly show for the player and non-synced units

v10.1.5.2764
	JULY 24, 2023 Hotfixes
	- Angel's Mercy's cooldown reduction is now 25% effective in PvP Combat.
	- Impish Instincts now reduces the cooldown of Demonic Circle by 3 seconds when triggered (was 2 seconds).
	Fixed an issue where Bestow Weyrnstone would persist when casted on another player
	Added Undulating Sporecloak, Well-Honed Insticts (passive procs)

v10.1.5.2763
	Symbol of Hope will correctly increase Obsidian Scales' cd recovery rate for Augmentation spec.

v10.1.5.2762
	Oppressing Roar w/ Overawe will correctly go on cooldown when used
	Overawe will correctly reduce Oppressing Roar's cooldown for each enrage effect dispelled
	Upheaval w/ Font of Magic will correctly go on cooldown when used

v10.1.5.2761
	Healbot fix

v10.1.5.2760
	Patch 10.1.5 updates

v10.1.0.2759
	nil err fix

v10.1.0.2758
	Fixed Rage of the Sleeper CD 90>60s
	Fixed Improved Mass Dispel CDR 25>20s
	Fixed Shadow Blades CD 180>120s

v10.1.0.2757
	Fixed Quick Witted CDR when interrupting Penance
	Fixed Invigorating Shadowdust CDR
	Fixed nil error on Flow State removal
	Fixed Ironbark CD 60>45s (Guardian, APRIL 5, 2023 hotfixes)

v10.1.0.2756
	JUNE 5, 2023 hotfixes
		Amplify Curse CD 30>60s
		Teachings of the Satyr CDR 10>15s
		Thunderlord CDR 1>1.5s
	MAY 15, 2023 hotfixes
		BM HUnter, Aberrus (4) Set bonus CDR 1>2s
		Sharpen Blade CD 25>30s

v10.1.0.2755
	CDR fix/updates
		Forbearance will no longer be affected by other CDR (e.g. Resolute Defender on Divine Shield)
		Fixed Uther's Councel CDR for 10.1 (rank1 20% to rank2 15%/30%)
		Fixed Inmost Light CDR for 10.1 (15% to 25%)
		Fixed Chi Surge CDR (was incorrectly capped at 16 sec instead of 20 sec)
		Updated Keg Smash CDR (was reducing Fortifying Brew by 5 sec instead of 3 sec. Bug fixed in 10.1)
		Updated Tiger Palm CDR (was reducing to all Brew CD by 0.5 sec instead of 1 sec. Bug fixed in 10.1)
	CD sync revamped
		Fixed an issue that caused certain cooldowns to sync improperly. Recipients now check remaining time over starting time
		Periodic sync has been removed and will now sync the followings directly like others
			% chance to incur no cooldown
			% chance to have its cooldown instantly reset when used
		Messages are now compressed and will no longer communicates with older versions
	10.1.5 compatibility updates
		Frames created while in combat will have tooltips disabled until you leave combat

v10.1.0.2753
	Bar position will correctly update in combat when using non-Group sorting in Arena
	Fixed Exhilaration and Trueshot CDR

v10.1.0.2752
	Max Number of Visible Icons will correctly work with Display Inactive Icons disabled
	Healing Stream Totem will correctly add a charge if you have both talents as Restoration spec
	Fixed Fortitude of the Bear CD

v10.1.0.2751
	Added Season 2 Obsidian PvP Trinkets
	Added Embers of Neltharion Tier Set bonuses
	Added option to use ElvUI's timer (in General menu)
	Fixed spell tooltips showing a blank text if the spell hasn't been cached yet
	Fixed option panel dropdown menu tooltip not showing

v10.1.0.2750
	Patch 10.1 updates
	Blessing of the Bronze CDR added to non-Evoker classes

v10.0.7.2748
	Fixed CDR to Incarn: Guardian of Ursoc
	nil error fix (nil mapInfo, nil info on sorting extraBars)
	Added ElvUI Raid Wide Sorting

v10.0.7.2747
	Added option to limit the number of visible icons per unit
	Added option to move icon names vertically
	Added option to scale status bar names
	Added option to rename Extra Bars
	Fixed Divine Protection for Rets
	Fixed Divine Purpose procs for Rets (HoJ CDR)
	Fixed Fire Breath CD reduction for Preservation
	Fixed anchor visibility not updating on zone change
	Visibility by Group Size can be configured for each zone separately
	Empowered spell will reset if the cast fails (simplified)
	Lib updates - cell padding, flash button fix, dropdown fix

	Hotfixes
	MARCH 28, 2023
	Holy Priest
	Fixed a bug causing Symbol of Hope to reduce the cooldown of both Divine Protection and Shield of Vengeance for Retribution
	Paladins. It will now only reduce the cooldown of Divine Protection.

	10.1 PTR fixes

v10.0.7.2746
	Cloudburst Totem will correctly replace Healing Stream Totem
	Casting Echo will correctly interact with Just in Time (Talent)
	Tooltip frames now uses a generic frames other addons can modify
	Removed Util\TooltipID (moved to dev ver)
	Cleaned up OmniCC support
	Removed ElvUI CD text support
	CD timers will explicitly use Blizzard's CD text
	ElvUI user can safely? enable Blizzard's CD text as WAs no longer show duplicate text
	Fixed lib load order

v10.0.7.2745
	Patch 10.0.7 spell & modifier updates
	Fixed Forbearance gray out effect incorrectly showing on Lay on Hands when it's used
	Added additional sorting for the Raid Bar
	Added UseIconAlpha for the Raid Bar
	Interrupt spell types are no longer visible in the Raid CD tab
	Addon will no longer sync with older versions (inspection will continue to work)

	Lib updates
	Battle Res Plugin:
		Fixed Arclight Vital Connector (DF engineering) not consuming a brez charge instantly

v10.0.5.2744

Altoholic/Core.lua
local addonName = ...

_G[addonName] = LibStub("AceAddon-3.0"):NewAddon(addonName, "AceConsole-3.0", "AceEvent-3.0", "AceComm-3.0", "AceSerializer-3.0", "AceTimer-3.0", "LibMVC-1.0")

local addon = _G[addonName]

addon.Version = "v1.13.017"
addon.VersionNum = 113017

local L = LibStub("AceLocale-3.0"):GetLocale(addonName)
local commPrefix = addonName

BINDING_HEADER_ALTOHOLIC = addonName
BINDING_NAME_ALTOHOLIC_TOGGLE = "Toggle UI"

local options = { 
	type = "group",
	args = {
		search = {
			type = "input",
			name = L['search'],
			usage = "<item name>",
			desc = L["Search in bags"],
			get = false,
			set = "CmdSearchBags",
		},
		show = {
			type = "execute",
			name = L['show'],
			desc = L["Shows the UI"],
			func = function() AltoholicFrame:Show() end
		},
		hide = {
			type = "execute",
			name = L['hide'],
			desc = L["Hides the UI"],
			func = function() AltoholicFrame:Hide() end
		},
		toggle = {
			type = "execute",
			name = L['toggle'],
			desc = L["Toggles the UI"],
			func = function() addon:ToggleUI() end
		},
	},
}
 
local AddonDB_Defaults = {
	global = {
		Guilds = {
			['*'] = {			-- ["Account.Realm.Name"] 
				hideInTooltip = nil,		-- true if this guild should not be shown in the tooltip counters
			},
		},
		Characters = {
			['*'] = {					-- ["Account.Realm.Name"] 
			},
		},
		Sharing = {
			Clients = {},
			SharedContent = {			-- lists the shared content
				--	["Account.Realm.Name"]  = true means the char is shared,
				--	["Account.Realm.Name.Module"]  = true means the module is shared for that char
			},
			Domains = {
				['*'] = {			-- ["Account.Realm"] 
					lastSharingTimestamp = nil,	-- a date, the last time information from this realm/account was queried and successfully saved.
					lastUpdatedWith = nil,		-- last player with whom the account sharing took place
				},
			},
		},
		unsafeItems = {},
		options = {
			-- ** Summary tab options **
			["UI.Tabs.Summary.ShowRestXP150pc"] = false,						-- display max rest xp in normal 100% mode or in level equivalent 150% mode ?
			["UI.Tabs.Summary.CurrentMode"] = 1,								-- current mode (1 = account summary, 2 = bags, ...)
			["UI.Tabs.Summary.CurrentColumn"] = "Name",						-- current column (default = "Name")
			["UI.Tabs.Summary.CurrentRealms"] = 2,								-- selected realms (current/all in current/all accounts)
			["UI.Tabs.Summary.CurrentFactions"] = 3,							-- 1 = Alliance, 2 = Horde, 3 = Both
			["UI.Tabs.Summary.CurrentLevels"] = 1,								-- 1 = All
			["UI.Tabs.Summary.CurrentLevelsMin"] = 1,							
			["UI.Tabs.Summary.CurrentLevelsMax"] = 60,					
			["UI.Tabs.Summary.CurrentClasses"] = 0,							-- 0 = All
			["UI.Tabs.Summary.CurrentTradeSkill"] = 0,						-- 0 = All
			["UI.Tabs.Summary.SortAscending"] = true,							-- ascending or descending sort order
			["UI.Tabs.Summary.ShowLevelDecimals"] = true,					-- display character level with decimals or not
			["UI.Tabs.Summary.ShowILevelDecimals"] = true,					-- display character level with decimals or not
			
			-- ** Character tab options **
			["UI.Tabs.Characters.ViewBags"] = true,
			["UI.Tabs.Characters.ViewBank"] = true,
			["UI.Tabs.Characters.ViewBagsAllInOne"] = false,
			["UI.Tabs.Characters.ViewBagsRarity"] = 0,						-- rarity level of items (not a boolean !)
			["UI.Tabs.Characters.SortAscending"] = true,						-- ascending or descending sort order
						
			-- ** Search tab options **
			["UI.Tabs.Search.ItemInfoAutoQuery"] = false,
			["UI.Tabs.Search.IncludeNoMinLevel"] = true,				-- include items with no minimum level
			["UI.Tabs.Search.IncludeMailboxItems"] = true,
			["UI.Tabs.Search.IncludeKnownRecipes"] = true,
			["UI.Tabs.Search.SortAscending"] = true,							-- ascending or descending sort order
			TotalLoots = 0,					-- make at least one search in the loot tables to initialize these values
			UnknownLoots = 0,
			
			-- ** Guild Bank tab options **
			["UI.Tabs.Guild.BankItemsRarity"] = 0,								-- rarity filter in the guild bank tab
			["UI.Tabs.Guild.BankAutoUpdate"] = false,							-- can the guild bank tabs update requests be answered automatically or not.
			["UI.Tabs.Guild.SortAscending"] = true,							-- ascending or descending sort order
			
			-- ** Grids tab options **
			["UI.Tabs.Grids.Reputations.CurrentXPack"] = 1,					-- Current expansion pack 
			["UI.Tabs.Grids.Reputations.CurrentFactionGroup"] = 1,		-- Current faction group in that xpack
			["UI.Tabs.Grids.Currencies.CurrentTokenType"] = nil,			-- Current token type (default to nil = all-in-one)  
			["UI.Tabs.Grids.Tradeskills.CurrentXPack"] = 1,					-- Current expansion pack 
			["UI.Tabs.Grids.Tradeskills.CurrentTradeSkill"] = 1,			-- Current tradeskill index
			["UI.Tabs.Grids.Dungeons.CurrentXPack"] = 1,						-- Current expansion pack 
			["UI.Tabs.Grids.Dungeons.CurrentRaids"] = 1,						-- Current raid index 

			-- ** Tooltip options **
			["UI.Tooltip.ShowItemSource"] = true,
			["UI.Tooltip.ShowItemCount"] = true,
			["UI.Tooltip.ShowSimpleCount"] = false,				-- display just the counter, without details (like AH, equipped, etc..)
			["UI.Tooltip.ShowTotalItemCount"] = true,
			["UI.Tooltip.ShowKnownRecipes"] = true,
			["UI.Tooltip.ShowItemID"] = false,						-- display item id & item level in the tooltip (default: off)
			["UI.Tooltip.ShowGatheringNodesCount"] = true,		-- display counters when mousing over a gathering node (default:  on)
			["UI.Tooltip.ShowCrossFactionCount"] = true,			-- display counters for both factions on a pve server
			["UI.Tooltip.ShowMergedRealmsCount"] = true,			-- display counters for characters on connected realms
			["UI.Tooltip.ShowAllAccountsCount"] = true,			-- display counters for all accounts on the same realm
            ["UI.Tooltip.ShowSellPrice"] = true,					-- Sell price per unit
			
			-- ** Mail options **
			["UI.Mail.GuildMailWarning"] = true,					-- be informed when a guildie sends a mail to one of my alts
			["UI.Mail.AutoCompleteRecipient"] = true,				-- Auto complete recipient name when sending a mail
			["UI.Mail.LastExpiryWarning"] = 0,						-- Last time a mail expiry warning was triggered
			["UI.Mail.TimeToNextWarning"] = 3,						-- Time before the warning is repeated ('3' = no warning for 3 hours)
			
			-- ** Minimap options **
			["UI.Minimap.ShowIcon"] = true,
			["UI.Minimap.IconAngle"] = 180,
			["UI.Minimap.IconRadius"] = 78,
			
			-- ** Calendar options **
			["UI.Calendar.WarningsEnabled"] = true,
			["UI.Calendar.UseDialogBoxForWarnings"] = false,	-- use a dialog box for warnings (true), or default chat frame (false)
			["UI.Calendar.WeekStartsOnMonday"] = false,

			WarningType1 = "30|15|10|5|4|3|2|1",		-- for profession cooldowns
			WarningType2 = "30|15|10|5|4|3|2|1",		-- for dungeon resets
			WarningType3 = "30|15|10|5|4|3|2|1",		-- for calendar events
			WarningType4 = "30|15|10|5|4|3|2|1",		-- for item timers (like mysterious egg)
			
			-- ** Global options **
			["UI.AHColorCoding"] = true,					-- color coded recipes at the AH
			["UI.VendorColorCoding"] = true,				-- color coded recipes at vendors
			["UI.AccountSharing.IsEnabled"] = false,	-- account sharing communication handler is disabled by default
			
			["UI.Scale"] = 1.0,
			["UI.Transparency"] = 1.0,
			["UI.ClampWindowToScreen"] = false,

		},
	}
}

addon.Colors = {
	white	= "|cFFFFFFFF",
	red = "|cFFFF0000",
	darkred = "|cFFF00000",
	green = "|cFF00FF00",
	orange = "|cFFFF7F00",
	yellow = "|cFFFFFF00",
	gold = "|cFFFFD700",
	teal = "|cFF00FF9A",
	cyan = "|cFF1CFAFE",
	lightBlue = "|cFFB0B0FF",
	battleNetBlue = "|cff82c5ff",
	grey = "|cFF909090",
	
	-- classes
	classMage = "|cFF69CCF0",
	classHunter = "|cFFABD473",
	
	-- recipes
	recipeGrey = "|cFF808080",
	recipeGreen = "|cFF40C040",
	recipeOrange = "|cFFFF8040",
	
	-- rarity : http://wow.gamepedia.com/Quality
	common = "|cFFFFFFFF",
	uncommon = "|cFF1EFF00",
	rare = "|cFF0070DD",
	epic = "|cFFA335EE",
	legendary = "|cFFFF8000",
	heirloom = "|cFFE6CC80",

	Alliance = "|cFF2459FF",
	Horde = "|cFFFF0000"
}

addon.Icons = {
	ready = "\124TInterface\\RaidFrame\\ReadyCheck-Ready:14\124t",
	waiting = "\124TInterface\\RaidFrame\\ReadyCheck-Waiting:14\124t",
	notReady = "\124TInterface\\RaidFrame\\ReadyCheck-NotReady:14\124t",
	questionMark = "Interface\\RaidFrame\\ReadyCheck-Waiting",

	Alliance = "Interface\\Icons\\INV_BannerPVP_02",
	Horde = "Interface\\Icons\\INV_BannerPVP_01",
	Neutral = "Interface\\Icons\\Achievement_character_pandaren_female",
}

-- ** LDB Launcher **
LibStub:GetLibrary("LibDataBroker-1.1"):NewDataObject(addonName, {
	type = "launcher",
	icon = "Interface\\Icons\\INV_Drink_13",
	OnClick = function(clickedframe, button)
		addon:ToggleUI()
	end,
	text = (Broker2FuBar) and addonName or nil,		-- only for fubar,  not for ldb
	label = addonName,
})



local guildMembersVersion = {} 	-- hash table containing guild member info

-- Message types
local MSG_SEND_VERSION							= 1	-- Send Altoholic's version
local MSG_VERSION_REPLY							= 2	-- Reply

-- *** Utility functions ***
local function GuildBroadcast(messageType, ...)
	local serializedData = addon:Serialize(messageType, ...)
	addon:SendCommMessage(commPrefix, serializedData, "GUILD")
end

local function GuildWhisper(player, messageType, ...)
	if DataStore:IsGuildMemberOnline(player) then
		local serializedData = addon:Serialize(messageType, ...)
		addon:SendCommMessage(commPrefix, serializedData, "WHISPER", player)
	end
end

local function SaveVersion(sender, version)
	guildMembersVersion[sender] = version
end

-- *** Guild Comm ***
local function OnAnnounceLogin(self, guildName)
	GuildBroadcast(MSG_SEND_VERSION, addon.Version)
end

local function OnSendVersion(sender, version)
	if sender ~= UnitName("player") then								-- don't send back to self
		GuildWhisper(sender, MSG_VERSION_REPLY, addon.Version)		-- reply by sending my own version
	end
	SaveVersion(sender, version)											-- .. and save it
end

local function OnVersionReply(sender, version)
	SaveVersion(sender, version)
end

local GuildCommCallbacks = {
	[MSG_SEND_VERSION] = OnSendVersion,
	[MSG_VERSION_REPLY] = OnVersionReply,
}

function addon:OnInitialize()
	addon.db = LibStub("AceDB-3.0"):New(addonName .. "DB", AddonDB_Defaults)
	LibStub("AceConfig-3.0"):RegisterOptionsTable(addonName, options)

	addon:RegisterChatCommand("Altoholic", "ChatCommand")
	addon:RegisterChatCommand("Alto", "ChatCommand")
	
	DataStore:SetGuildCommCallbacks(commPrefix, GuildCommCallbacks)

	addon:RegisterMessage("DATASTORE_ANNOUNCELOGIN", OnAnnounceLogin)
	
	addon:RegisterComm("AltoShare", "AccSharingHandler")
	addon:RegisterComm(commPrefix, DataStore:GetGuildCommHandler())
	
	-- this event MUST stay here, we have to be able to respond to a request event if the guild tab is not loaded
	addon:RegisterMessage("DATASTORE_GLOBAL_MAIL_EXPIRY")
	addon:RegisterMessage("DATASTORE_CS_TIMEGAP_FOUND")
end

function addon:GetGuildMemberVersion(member)
	if guildMembersVersion[member] then			-- version number of a main ?
		return guildMembersVersion[member]		-- return it immediately
	end
	
	-- check if member is an alt
	local main = DataStore:GetNameOfMain(member)
	if main and guildMembersVersion[main] then
		return guildMembersVersion[main]
	end
end

function addon:ChatCommand(input)
	if not input then
		LibStub("AceConfigDialog-3.0"):Open(addonName)
	else
		LibStub("AceConfigCmd-3.0").HandleCommand(addon, "Alto", "Altoholic", input)
	end
end

local SPELL_ID_ALCHEMY = 2259
local SPELL_ID_BLACKSMITHING = 3100
local SPELL_ID_ENCHANTING = 7411
local SPELL_ID_ENGINEERING = 4036
local SPELL_ID_LEATHERWORKING = 2108

local SPELL_ID_TAILORING = 3908
local SPELL_ID_SKINNING = 8613
local SPELL_ID_MINING = 2575
local SPELL_ID_HERBALISM = 2366
-- local SPELL_ID_SMELTING = 2656

local SPELL_ID_COOKING = 2550
local SPELL_ID_FIRSTAID = 3273
local SPELL_ID_FISHING = 7733

addon.TradeSkills = {
	Recipes = {},
	-- spell IDs in alphabetical order (english), primary then secondary
    spellIDs = { 2259, 3100, 7411, 4036, 2108, 3908, 8613, 2575, 2366, 2550, 3273, 7733 },
	firstSecondarySkillIndex = 10, -- index of the first secondary profession in the table
	
	AccountSummaryFiltersSpellIDs = { 2259, 3100, 7411, 4036, 2108, 3908, 8613, 2575, 2366, 2550, 3273, 7733 },
	AccountSummaryFirstSecondarySkillIndex = 10, -- index of the first secondary profession in the table
		
	Names = {
		ALCHEMY = GetSpellInfo(2259),
		BLACKSMITHING = GetSpellInfo(3100),
		COOKING = GetSpellInfo(2550),
		ENCHANTING = GetSpellInfo(7411),
		ENGINEERING = GetSpellInfo(4036),
        FIRSTAID = GetSpellInfo(3273),
		FISHING = GetSpellInfo(18248),
		HERBALISM = GetSpellInfo(2366),
		LEATHERWORKING = GetSpellInfo(2108),
		MINING = GetSpellInfo(2575),
		SKINNING = GetSpellInfo(8613),
		TAILORING = GetSpellInfo(3908),
	},
}

-- ** Tabs **
local tabList = {
	"Summary",
	"Characters",
	"Search",
	"Guild",
	"Grids",
}

local frameToID = {}
for index, name in ipairs(tabList) do
	frameToID[name] = index
end

local function SafeLoadAddOn(name)
	if not IsAddOnLoaded(name) then
		LoadAddOn(name)
	end
end

local function ShowTab(name)
	local tab = _G[addonName.."Tab" .. name]
	if tab then
		tab:Show()
	end
end

local function HideTab(name)
	local tab = _G[addonName.."Tab" .. name]
	if tab then
		tab:Hide()
	end
end

addon.Tabs = {}

function addon.Tabs:HideAll()
	for _, tabName in pairs(tabList) do
		HideTab(tabName)
	end
end

function addon.Tabs:OnClick(index)
	if type(index) == "string" then
		index = frameToID[index]
	end
	
	PanelTemplates_SetTab(_G[addonName.."Frame"], index);
	self:HideAll()
	self.current = index
	
	if index >= 1 and index <= 7 then
		local moduleName = format("%s_%s", addonName, tabList[index])
		SafeLoadAddOn(moduleName)		-- make this part a bit more generic once we'll have more LoD parts
		
		local parentLevel = AltoholicFrame:GetFrameLevel()
		local childName = format("%sTab%s", addonName, tabList[index])

		local tabFrame = _G[ childName ]
		
		if tabFrame then
			local childLevel = tabFrame:GetFrameLevel()
			
			if childLevel <= parentLevel then	-- if for any reason a child frame has a level lower or equal to its parent, fix it
				tabFrame:SetFrameLevel(parentLevel+1)
			end
		else
			addon:Print(format("%s is disabled.", moduleName))
		end
	end
	
	ShowTab(tabList[index])
end

-- Allow ESC to close the main frame
tinsert(UISpecialFrames, "AltoholicFrame")
tinsert(UISpecialFrames, "AltoMessageBox")

function addon:CmdSearchBags(arg1, arg2)
	-- arg 1 is a table, no idea of what it does, investigate later, only  arg2 matters at this point
	
	if string.len(arg2) == 0 then
		addon:Print("|cFF00FF9A" .. L["Altoholic:|r Usage = /altoholic search <item name>"])
		return
	end
	
	if not (AltoholicFrame:IsVisible()) then
		AltoholicFrame:Show()
	end
	AltoholicFrame_SearchEditBox:SetText(strlower(arg2))
	addon.Tabs:OnClick("Search")
	addon.Search:FindItem()
end	
